scikeras/wrappers.py

```diff
@@ -533,20 +605,17 @@ def _fit(self, X, y, sample_weight=None, warm_start=False):
             ValueError : In case of invalid shape for `y` argument.
         """
         # Handle random state
-        if hasattr(self, "random_state"):
-            if isinstance(self.random_state, np.random.RandomState):
-                # Keras needs an integer
-                # we sample an integer and use that as a seed
-                # Given the same RandomState, the seed will always be
-                # the same, thus giving reproducible results
-                state = self.random_state.get_state()
-                self._random_state = self.random_state.randint(low=1)
-                self.random_state.set_state(state)
-            else:
-                # int or None
-                self._random_state = self.random_state
+        if isinstance(self.random_state, np.random.RandomState):
```

Comment:
``` python
from sklearn.utils import check_random_state

rng = check_random_state(self.random_state)
assert isinstance(rng, np.random.RandomState)
```

Reply:
I'm not sure what you're implying here.

We don't use `check_random_state` because we want the opposite behavior of most sklearn estimators: they accept an int or RandomState instance, but want a RandomState instance (which is what `check_random_state` is for). We accept and int or a RandomState instance, but we want an int.

Reply:
Touche. I'll resolve this conversation.